src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,56 @@
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+import { FilterParams } from '../types/FilterParams';
+import { CENTURIES } from '../utils/centuries';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const queryFilter = searchParams.get(FilterParams.Query) || '';
+  const sexFilter = searchParams.get(FilterParams.Sex) || '';
+  const selectedCenturies = searchParams.getAll(FilterParams.Centuries) || [];
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const updatedSearchParams = getSearchWith(searchParams, {
+      query: event.target.value || null,
+    });
+
+    setSearchParams(updatedSearchParams);
+  };
+
+  const handleCenturyChange = (currentCentury: string) => {
+    return selectedCenturies.includes(currentCentury)
+      ? selectedCenturies.filter((century) => century !== currentCentury)
+      : [...selectedCenturies, currentCentury];
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <SearchLink
+          params={{ sex: null }}
+          className={classNames({ 'is-active': !sexFilter })}
+        >
+          All
+        </SearchLink>
+
+        <SearchLink
+          params={{ sex: 'm' }}
```

Comment:
Consider will be better if you create a enum for these magic words. and use them everywhere